jetty-ee9/jetty-ee9-webapp/src/main/java/org/eclipse/jetty/ee9/webapp/WebAppClassLoader.java

```diff
@@ -112,7 +113,7 @@ public interface Context extends ClassVisibilityChecker
          */
         boolean isParentLoaderPriority();
 
-        ResourceCollection getExtraClasspath();
+        Resource getExtraClasspath();
```

Comment:
@lorban @joakime @sbordet  I believe it was a miss-use of `ResourceCollection` when this was changed from a `List<Resource>`.    We should very VERY rarely ever explicitly use a `ResourceCollection`, because the whole point of that class is to look like a singe resource.   If anything needs to iterate over the contained resources, then it probably should be using a `List<Resources>` rather than a `ResourceCollection`.

However, for not, I have only partially reverted to try how it looks if extra classpath is just a `Resource`, which may or may not be a collection.   This requires the `addClassPath` method below to be somewhat `ResourceCollection` away, as it needs to extract URLs.  I've done this for now with a static stream method.... but not sure I like.   Reverting all the way back to `List<Resource>` may be the best thing to do here.

Reply:
maybe `ResourceCollection` should be renamed to `MultiResource` to emphasise that it is-a `Resource` and should not be thought of as a `Collection` (which is an implementation detail).

Reply:
Better name options include:
 + `MultiResource`
 + `CompositeDirectoryResource`
 + `CombinedResource`
 + `CombinedDirectoryResource`